llvm/lib/Target/WebAssembly/WebAssemblyInstrControl.td

```diff
@@ -147,6 +178,8 @@ let isTerminator = 1, hasSideEffects = 1, isBarrier = 1, hasCtrlDep = 1,
 // usage gets low enough.
 
 // Rethrowing an exception: rethrow
+// The new exnref proposal also uses this instruction as an interim pseudo
```

Comment:
If we want to be strict maybe we can make the old rethrow `RETHROW_LEGACY` or something too and add a new pseudo `RETHROW`, but I didn't feel it very necessary given that this is not very confusing. (In `CATCH`/`CATCH_ALL`'s case it was confusing because in CFGStackify or MCInstLower we have to check `WebAssembly::WasmEnableExnref` many times to check whether the current instruction is the legacy `CATCH` or the new pseudo `CATCH`)

Reply:
I think I see; before LateEHPrepare, there is no exnref that comes out of the catch, so it doesn't make sense to have a throw_ref either. So instead we have rethrow, which has semantics that are meaningful on their own in this phase of compilation even though they don't correspond to what we need at the end.
And yeah, given that rethrow is a meaningful concept independent of the legacy spec, I don't think we need to try to keep it labeled as legacy permanently.